### DISCUSSION TITLE
Support for non-bootable image

### DISCUSSION THREAD

**Discussion by PetrBuchtaNXP** [mcu-tools/mcuboot#1393] [Q&A] [answered]
I have a use-case where I need mcuboot to only boot one image that functions as main application. However, there are other images that are not supposed to be loaded by mcuboot, only managed during their update (application data, firmware for other computing cores, etc.).

I noticed in the code some use of IMAGE_F_NON_BOOTABLE definition.  However, I could not find a way with imgtool to enforce this flag when signing an image. Could someone explain what is main purpose of this attribute?

Also, is there a way how to skip verification of non-bootable image in case it's not present in flash? I find this useful in case I have an image that functions as optional application data that is not always needed but when it is present I would like its signature to be verified.

Thanks for clarification.

**Comment by nvlsianpu** [comment, marked as answer]
looks like  IMAGE_F_NON_BOOTABLE is an inheritance from mynewt bootloader. Doesn't look as something you want to use.

If multi-image update is the mode you can use: If non of others image depends on the image which is not provided, update will be performed.
If the image will be provided - it will be verified.
